pubsub/gossipsub/gossipsub-v1.1.md

```diff
@@ -634,6 +637,83 @@ In order counter spam that elicits responses and consumes resources, some measur
   reducing the surface of spam-induced computation (eg validation). The application can take
   further steps and blacklist the peer if the spam persists after the negative score decays.
 
+### Validation Queue Protection
+
+An important aspect of gossipsub is the reliance on validators to signal acceptance of incoming
+messages from the application to the router. The validation is asynchronous, with a typical
+implementation strategy that uses of a front-end queue and a limit to the number of ongoing validations.
+This creates a potential target for attacks, as an attacker can overload the queue by brute force,
+sending spam messages at a very high rate. The effect would be that legitimate messages get dropped
+by the validation front end, resulting in denial of service.
+
+In order to protect the system from this class of attacks, gossipsub v1.1 incorporates a circuit
+breaker that sits before the validation queue and can make informed decisions on whether to
+push a message into the validation queue. This defensive mechanism kicks in when the system detects
+an elevated rate of dropped messages, and makes decisions on whether to accept incoming messages for
+validation based on the statistical performance of peers in the origin IP address. The decision is
+probabilistic and implements a Random Early Drop (RED) strategy that drops messages with a probability
+that depends on the acceptance rates for messages from the origin IP. This strategy can neuter
+attacks on the validation queue, because messages are no longer dropped indiscrimantly in a drop-tail
```

Comment:
editorial nit

```suggestion
attacks on the validation queue, because messages are no longer dropped indiscriminately in a drop-tail
```

Reply:
ugh, thanks for being a good spell checker!

Reply:
fixed!